nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -1152,9 +1158,13 @@ def modified_adaptive_expansion_search(
                 beam_dec_out = torch.stack([h.dec_out[-1] for h in hyps])  # [H, 1, D]
 
                 # Extract the log probabilities
-                beam_logp, beam_idx = torch.log_softmax(
-                    self.joint.joint(beam_enc_out, beam_dec_out) / self.softmax_temperature, dim=-1,
-                ).topk(self.max_candidates, dim=-1)
+                if isinstance(self.joint, hat.HATJoint) and self.hat_subtract_ilm:
```

Comment:
Here, everywhere, simply call the self.joint.joint(with the ordinary arguments for RNNT). The output can now be either torch.Tensor - (RNNT joint, HAT without the ILM subtract) or it can be HATOutput dataclass. 

import RNNT utils and then check if torch.is_tensor(output) here - this is for og RNNT. Elif self.hat_subtract_ilm and isinstance(output, HATOutput): 

Then do the required code path. On else path, raise error saying could not resolve the output